src/modules/rover_pos_control/RoverPositionControl.cpp

```diff
@@ -265,6 +265,37 @@ RoverPositionControl::control_position(const matrix::Vector2f &current_position,
 	return setpoint;
 }
 
+bool
+RoverPositionControl::control_velocity(const matrix::Vector2f &current_velocity,
+				       const matrix::Vector3f &ground_speed, const position_setpoint_triplet_s &pos_sp_triplet)
+{
+	bool setpoint = true;
+	float mission_throttle = _param_throttle_cruise.get();
+	matrix::Vector2f desired_velocity{pos_sp_triplet.current.vx, pos_sp_triplet.current.vy};
+	float desired_speed = desired_velocity.norm();
+
+	if (desired_speed > 0.01f) {
+
+		float control_throttle = _act_controls.control[actuator_controls_s::INDEX_THROTTLE] +
+					 (desired_velocity.norm() - current_velocity.norm());
+		_act_controls.control[actuator_controls_s::INDEX_THROTTLE] = math::constrain(control_throttle, 0.0f, mission_throttle);
+
+		float desired_theta = atan2f(desired_velocity(1), desired_velocity(0)) - atan2f(current_velocity(1),
+				      current_velocity(0));
```

Comment:
Is there a chance this will return `NaN`?

Reply:
Makes sense thanks!

Reply:
I think if the current_velocity is nan, this might be the case. desired_velocity I believe it will be checked on the mavlink_receiver side wheter it is a `NaN`. I am not sure what is a good way to handle this. Could you give me any pointers?

Reply:
@julianoes I have fixed the way we calculate the desired angle to steer. Think it is better to only look at the desired_velocity in the body frame